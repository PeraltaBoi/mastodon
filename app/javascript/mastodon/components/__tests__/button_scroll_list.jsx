import renderer from 'react-test-renderer';

//import { render, fireEvent } from 'mastodon/test_helpers';

import ButtonScrollList from '../button_scroll_list';

jest.mock('mastodon/components/icon', () => 'Icon');

describe('<ButtonScrollList />', () => {
  it('renders an empty button scroll list element', () => {
    const children = [];
    const component = renderer.create(
      <ButtonScrollList>{children}</ButtonScrollList>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders the children', () => {
    const children = Array.from({ length: 5 }, (_, i) => (
      <div key={i} ref={jest.fn()} />
    ));
    const component = renderer.create(
      <ButtonScrollList>{children}</ButtonScrollList>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('scrolls left', () => {
    const children = Array.from({ length: 5 }, (_, i) => (
      <div key={i} ref={jest.fn()} />
    ));
    const component = renderer.create(
      <ButtonScrollList>{children}</ButtonScrollList>,
    );
    const instance = component.getInstance();
    instance.scrollLeft();
  });

  it('scrolls right', () => {
    const children = Array.from({ length: 5 }, (_, i) => (
      <div key={i} ref={jest.fn()} />
    ));
    const component = renderer.create(
      <ButtonScrollList>{children}</ButtonScrollList>,
    );
    const instance = component.getInstance();
    instance.scrollRight();
  });
});