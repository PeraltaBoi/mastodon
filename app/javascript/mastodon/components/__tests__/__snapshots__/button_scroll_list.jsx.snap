// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonScrollList /> renders an empty button scroll list element 1`] = `
<div
  className="button-scroll-list-container"
>
  <button
    className="icon-button column-header__setting-btn"
    onClick={[Function]}
  >
    <div
      aria-hidden={true}
      className="icon icon-chevron-left"
      title=""
    />
  </button>
  <div
    className="button-scroll-list"
  />
  <button
    className="icon-button column-header__setting-btn"
    onClick={[Function]}
  >
    <div
      aria-hidden={true}
      className="icon icon-chevron-right"
      title=""
    />
  </button>
</div>
`;

exports[`<ButtonScrollList /> renders the children 1`] = `
<div
  className="button-scroll-list-container"
>
  <button
    className="icon-button column-header__setting-btn"
    onClick={[Function]}
  >
    <div
      aria-hidden={true}
      className="icon icon-chevron-left"
      title=""
    />
  </button>
  <div
    className="button-scroll-list"
  >
    <div>
      <div />
    </div>
    <div>
      <div />
    </div>
    <div>
      <div />
    </div>
    <div>
      <div />
    </div>
    <div>
      <div />
    </div>
  </div>
  <button
    className="icon-button column-header__setting-btn"
    onClick={[Function]}
  >
    <div
      aria-hidden={true}
      className="icon icon-chevron-right"
      title=""
    />
  </button>
</div>
`;
